zerver/views/registration.py

```diff
@@ -327,11 +334,13 @@ def accounts_register(request: HttpRequest) -> HttpResponse:
 
         if user_profile is None:
             user_profile = do_create_user(email, password, realm, full_name, short_name,
+                                          user_avatar_url=user_avatar_url,
                                           prereg_user=prereg_user,
                                           is_realm_admin=is_realm_admin,
                                           is_guest=is_guest,
                                           tos_version=settings.TOS_VERSION,
                                           timezone=timezone,
+                                          avatar_source=avatar_source,
```

Comment:
Do we need to pass `avatar_source` here?  I think its value is implied by `user_avatar_url` being passed.

Reply:
Now I think yes because we can have a `user_avatar_url` value, but if the user doesn't choose to use their Github avatar, then the source will be Gravatar instead of from user.